### DISCUSSION TITLE
How to handle multipart/form-data

### DISCUSSION THREAD

**Discussion by rizchaerul** [TypedProject/ts-express-decorators#1716] [Q&A] [answered]
Hi, I read the docs about multipart/form-data (https://tsed.io/api/platform/common/types/decorators/multer/MultipartFile.html)

But it seems that it only support file field? Is it possible if I declare it like this?
![image](https://user-images.githubusercontent.com/72289631/147956417-08cb650b-40a7-4b60-a260-3b2241d105fc.png)

If not then what would you recommend to get all the field + files from multipart/form-data.

Thank you

**Comment by Romakita** [comment, marked as answer]
Hello @rizchaerul 

You have to use the `@MultipartFile` to get the file. @BodyParams isn't designed for that. But you can use BodyParams decorator to get extra fields.

```typescript
get(@MultipartFile('file') file: any, @BodyParams() form: AppForm)
```

See you
Romain

**Reply by rizchaerul** [reply]
Tried that and it worked, thanks!
